pennylane/interfaces/batch/autograd.py

```diff
@@ -212,27 +205,20 @@ def grad_fn(dy):
                         )
                     )
 
-            elif (
-                hasattr(gradient_fn, "fn")
-                and inspect.ismethod(gradient_fn.fn)
-                and gradient_fn.fn.__self__ is device
-            ):
-                # Gradient function is a device method.
+            else:
+                # Gradient function is not a gradient transform
+                # (e.g., it might be a device method).
                 # Note that unlike the previous branch:
                 #
                 # - there is no recursion here
                 # - gradient_fn is not differentiable
                 #
                 # so we cannot support higher-order derivatives.
-
                 with qml.tape.Unwrap(*tapes):
                     jacs = gradient_fn(tapes, **gradient_kwargs)
 
                 vjps = [qml.gradients.compute_vjp(d, jac) for d, jac in zip(dy, jacs)]

```

Comment:
The similar case in line 193 was indented to sit within the `with` statement here; should the same be true here?

Reply:
I don't believe so, no. I can't say why though, but the tests don't seem to error here 🤔